metrics/sanitize.go

```diff
@@ -0,0 +1,47 @@
+package metrics
+
+import (
+	"strings"
+	"unicode"
+)
+
+// SanitizeLabel sanitizes a label to conform with Prometheus label rules
+func SanitizeLabel(input string) string {
+	input = strings.TrimSpace(input)
+
+	if input == "" {
+		return input
+	}
+
+	var result strings.Builder
+
+	runeInput := []rune(input)
+	firstCh := runeInput[0]
+	// To prevent leading non-letter symbol
+	isAllowedLeadingCharacter := unicode.IsLetter(firstCh) || firstCh == '_'
```

Comment:
minor: `firstCh == '_'` can be removed here as well - the logic would be the same